### PR TITLE
PYTHON-957

### PR DESCRIPTION
**Pull request by jsamoocha**
Implemented rich comparison methods for write operations, as indicated in [PYTHON-957](https://jira.mongodb.org/browse/PYTHON-957).

### CODE REVIEW

File: pymongo/operations.py [hunk 1]
```diff
@@ -32,6 +32,15 @@ def __init__(self, filter=None, doc=None, upsert=None):
         self._doc = doc
         self._upsert = upsert
 
+    def __eq__(self, other):
+        if type(other) == type(self):
+            return (other._filter, other._doc, other._upsert) == \
+                   (self._filter, self._doc, self._upsert)
+        return False
```
Comment (behackett) [inline pymongo/operations.py]: This should return NotImplemented, instead of False.

Comment (behackett) [inline pymongo/operations.py]: Oh, I see what you are doing here. You want to return NotImplemented if the comparison is to a type that doesn't inherit from _WriteOp and False if the type doesn't match.

Comment (jsamoocha) [inline pymongo/operations.py]: You're attributing too much sophistication to my code :-), this was my first attempt at implementing `__eq__()` in Python and I simply wasn't aware of `NotImplemented` as a potential return value, so good catch there.

I changed the return value in a new commit - the behavior of `assertNotEqual()` in tests doesn't seem to be affected by this change.

**Comment by behackett** [comment]
Merged to master. Thanks for the patch!

**Pull request closed without merging**